lib/azure_text_to_speech.ex

```diff
@@ -17,18 +17,22 @@ defmodule AzureTextToSpeech do
     audio-48khz-96kbitrate-mono-mp3     audio-48khz-192kbitrate-mono-mp3)
   @user_agent "awesome"
 
+  @spec voices_list :: map()
   def voices_list do
     voices_list_endpoint()
     |> HTTPoison.get!(authorization_header(access_token()))
     |> Map.get(:body)
     |> Jason.decode!()
   end
 
+  @spec audio_list :: [bitstring(), ...]
   def audio_list do
     @audio_list
   end
 
-  def ssml(text, %{"Name" => name, "Locale" => locale, "Gender" => gender}) do
+  @spec ssml(bitstring(), map()) :: bitstring()
+  def ssml(text, %{"Name" => name, "Locale" => locale, "Gender" => gender} = voice)
+      when is_bitstring(text) and is_map(voice) do
```

Comment:
This `is_map` guard is redundant because the map type is checked by pattern matching.
```suggestion
      when is_binary(text) do
```